lib/lambda_ethereum_consensus/p2p/gossip/operations_collector.ex

```diff
@@ -56,6 +105,30 @@ defmodule LambdaEthereumConsensus.P2P.Gossip.OperationsCollector do
       state.bls_to_execution_change
       |> Enum.reject(&MapSet.member?(indices, &1.message.validator_index))
 
-    %{state | bls_to_execution_change: bls_to_execution_changes}
+    # TODO: improve AttesterSlashing filtering
+    attester_slashings =
+      state.attester_slashing |> Enum.reject(&Enum.member?(operations.attester_slashings, &1))
+
+    slashed_proposers =
+      operations.proposer_slashings |> MapSet.new(& &1.signed_header_1.message.proposer_index)
+
+    proposer_slashings =
+      state.proposer_slashing
+      |> Enum.reject(
+        &MapSet.member?(slashed_proposers, &1.signed_header_1.message.proposer_index)
+      )
+
+    exited = operations.voluntary_exits |> MapSet.new(& &1.message.validator_index)
+
+    voluntary_exits =
+      state.voluntary_exit |> Enum.reject(&MapSet.member?(exited, &1.message.validator_index))
+
+    %{
+      state
```

Comment:
not asking to change this now, but wonder if it would be better to have one GenServer per operation to benefit from parallelization.

Reply:
Yeah, probably. It depends on how slow the filtering is.